cpp/src/io/json/write_json.cu

```diff
@@ -254,38 +182,200 @@ std::unique_ptr<column> struct_to_strings(table_view const& strings_columns,
                            strings_columns.end(),
                            [](auto const& c) { return c.type().id() == type_id::STRING; }),
                "All columns must be of type string");
+  auto constexpr str_views_per_column = 3;  // (for each "column_name:", "value",  "separator")
+  auto const num_str_views_per_row    = strings_columns.num_columns() * str_views_per_column + 1;
+  // Eg. { col1: value , col2: value , col3: value } = 1+ 3+3+(3-1) +1 = 10
+
+  auto tbl_device_view = cudf::table_device_view::create(strings_columns, stream);
+  auto d_column_names  = column_device_view::create(column_names, stream);
+
+  // (num_columns*3+1)*num_rows size (very high!) IDEA: chunk it here? but maximize parallelism?
+  auto const string_views_total = num_str_views_per_row * strings_columns.num_rows();
+  auto const total_rows         = strings_columns.num_rows() * strings_columns.num_columns();
+  using str_pair                = thrust::pair<const char*, size_type>;
```

Comment:
Using `thrust::pair` but calling these string-views irks me a bit. Perhaps use string-pair or some other name for these?

Reply:
There is a string-view version of the factory too if you wanted to use `string_view`s instead of `std::pair`. Was there something preventing you from using that one?
https://github.com/rapidsai/cudf/blob/267bc6a82dd7ddd241e2d722af08e2f5bf996e6f/cpp/include/cudf/column/column_factories.hpp#L407-L411

Reply:
It transforms to pairs. That's why I used pairs directly.

Reply:
used `string_view` itself. used `string_view` version of the factory too.